### ISSUE TITLE
Translation of Centre of Gravity to global origin

### ISSUE THREAD

**Issue by Vivian0108** [WoLpH/numpy-stl#60]
When center of gravity is calculated for a stl file, and the mesh is translated such that
my_mesh.translate(-1*cog), i.e entire mesh is translated such that its cog coincides with the origin. But when the cog of the translated mesh is calculated, it does not show origin, but rather a very small value such as ([ -1.02913186e-07,  -1.07319586e-07,  -2.17118228e-06]). So the mesh is not accurately translating to origin. Please check

**Comment by WoLpH** [comment]
Without your STL file I have no way to debug this unfortunately.

The COG method is tested and should do the trick: https://github.com/WoLpH/numpy-stl/blob/71958638f4684c50cf033aac71a293b897193c8f/tests/test_meshProperties.py

**Comment by Uvar** [comment]
Could you try adapting and running the following function on your mesh object? If the oriented area does not sum to 0 in all dimensions, your mesh is corrupted and the COG method will not work.

```
def area_or(self):
    v0 = self.data['vectors'][:, 0]
    v1 = self.data['vectors'][:, 1]
    v2 = self.data['vectors'][:, 2]
        
    normals = numpy.cross(v1 - v0, v2 - v0)
    normals_sum = numpy.array([math.fsum(normals[:, 0]),
                               math.fsum(normals[:, 1]),
                               math.fsum(normals[:, 2])])
    return normals_sum
```

**Comment by Uvar** [comment]
After mulling it over for a while I came to the conclusion that I misinterpreted the original issue. Although small mesh corruptions are not completely out of the picture, it seems more likely that OP wants floating point arithmetic to be fixed within-code. 
The best pseudo-solution I can currently think of is a recursive implementation with a user-specified error bound, but I am not sure the "error" is guaranteed to asymptotically decrease while that would be the base assumption for such a recursion (actually I am quite sure it is not at all guaranteed to).
